### PR TITLE
Add tm Org and VDC support

### PR DESCRIPTION
**Pull request by Didainius**
Changelog contains more details on the implemented structures.
`make testtm` passes.

### CODE REVIEW

File: govcd/tm_supervisor.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// wrap is a hidden helper that facilitates the usage of a generic CRUD function
+//
+//lint:ignore U1000 this method is used in generic functions, but annoys staticcheck
+func (s Supervisor) wrap(inner *types.Supervisor) *Supervisor {
+	s.Supervisor = inner
+	return &s
+}
+
+func (vcdClient *VCDClient) GetAllSupervisors(queryParameters url.Values) ([]*Supervisor, error) {
```
Comment (adambarreiro) [inline govcd/tm_supervisor.go:26]: Should we add docs to public methods?

**Review by adambarreiro — APPROVED** [review]
Tests passed for me

File: .changes/v3.0.0/710-features.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+* Added `TmOrg` and `types.TmOrg` to structure for OpenAPI management of Organizations with methods
+  `VCDClient.CreateTmOrg`, `VCDClient.GetAllTmOrgs`, `VCDClient.GetTmOrgByName`,
+  `VCDClient.GetTmOrgById`, `TmOrg.Update`, `TmOrg.Delete`, `.TmOrg.Disable` [GH-710]
+* Added `TmVdc` and `types.TmVdc` to structure for OpenAPI management of Organizations VDCs with
+  methods `VCDClient.CreateTmVdc`, `VCDClient.GetAllTmVdcs`, `VCDClient.GetTmVdcByName`,
+  `VCDClient.GetTmVdcById`, `TmVdc.Update`, `TmVdc.Delete`, `.TmVdc.Disable` [GH-710]
+* Added types `Region` and `types.Region` for managing regions with methods
+  `VCDClient.CreateRegion`, `VCDClient.GetAllRegions`, `VCDClient.GetRegionByName`,
+  `VCDClient.GetRegionById`, `Region.Update`, `Region.Delete` [GH-710]
```
Comment (lvirbalas) [inline .changes/v3.0.0/710-features.md:9]: I'm a bit hesitant to merge this PR until we can write tests for create/update/delete.

File: govcd/tm_org.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+//
+//lint:ignore U1000 this method is used in generic functions, but annoys staticcheck
+func (g TmOrg) wrap(inner *types.TmOrg) *TmOrg {
+	g.TmOrg = inner
+	return &g
+}
+
+func (vcdClient *VCDClient) CreateTmOrg(config *types.TmOrg) (*TmOrg, error) {
+	if !vcdClient.Client.IsTm() {
+		return nil, fmt.Errorf("err")
```
Comment (adambarreiro) [inline govcd/tm_org.go:31]: Just noticed this error message, looks incomplete?

**Comment by Didainius** [comment]
Splitting into pieces.

**Pull request closed without merging**